hatch-dependency-coversion/README.md

```diff
@@ -0,0 +1,76 @@
+# hatch-dependency-coversion
+
+[![PyPI - Version](https://img.shields.io/pypi/v/hatch-dependency-coversion.svg)](https://pypi.org/project/hatch-dependency-coversion)
+[![PyPI - Python Version](https://img.shields.io/pypi/pyversions/hatch-dependency-coversion.svg)](https://pypi.org/project/hatch-dependency-coversion)
+[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)
+[![Checked with mypy](http://www.mypy-lang.org/static/mypy_badge.svg)](http://mypy-lang.org/)
+
+-----
+
+This is a plugin for [Hatch](https://github.com/pypa/hatch) that allows you to rewrite the versions in selected dependency specifiers to be exactly the same as the current version of the project configured in `pyproject.toml`'s `[project]` table, requiring exact coversioning of those dependencies with the project.
+
+This is useful for projects that are developed in lockstep but distributed as independent python packages rather than as subcomponents of the same namespace package. It is the equivalent of the following `setup.py`:
+
+``` python
+VERSION = '1.0.0'
+setup(name='my-project', version=VERSION, install_requires=[f'my-dependency=={VERSION}'])
+```
+
+Minimal configuration is done in your `pyproject.toml` like this:
+
+``` toml
+# Since this is a plugin for hatchling, you must be using hatchling as your build backend 
+[build-system]
```

Comment:
The GitHub syntax highlighter seems to be choking on some of these comments, for some reason. I could fix it by moving the comments around. I'll push a fix directly.